packages/grafana-ui/src/components/Forms/Form.mdx

```diff
@@ -106,11 +108,38 @@ import { Forms } from '@grafana/ui';
   </>
 )}
 </Forms.Form>
+```
+Note that when using `Forms.InputControl`, it expects the name of the prop that handles input change to be called `onChange`.
+If the property is named differently for any specific component, additional `onChangeName` prop has to be provided, specifying the name.
+Additionally, the value to for the `onChange` prop is passed as an array. Check [react-hook-form docs](https://react-hook-form.com/api/#Controller)
+for more prop options.
+
+```jsx
+{/* In case of Select the value has to be returned as an object with a `value` key for the value to be saved to form data */}
+const onSelectChange = ([value]) => {
+  // ...
+  return { value };
+}
+
+<Field label="SelectExample">
+  <Forms.InputControl
+    as={Select}
```

Comment:
I'm not sure this example is a good one. When I look at it and see `as={Select}` I automatically think that this is grafana/ui Select (which isn't in this case as grafana/ui select does not have onSelectChange property). It's great you are providing this example, but would be even better, if it contained some dummy controlled component used instead of Select to picture the case you are describing here.

Reply:
Ah, I see what, you mean, good point! I was thinking about `DashboardPicker` when writing this, so I'd probably use that as an example.